src/main/groovy/com/github/benmanes/gradle/versions/updates/DependencyUpdatesTask.groovy

```diff
@@ -91,6 +98,18 @@ class DependencyUpdatesTask extends DefaultTask {
     this.resolutionStrategy = null
   }
 
+  void rejectVersionIf(final Transformer<Boolean, ComponentSelectionWithCurrent> filter) {
+    resolutionStrategy { ResolutionStrategyWithCurrent strategy ->
```

Comment:
Sounds like you can now drop the these now that the delegation is fixed. Does it work like this?

```gradle
resolutionStrategy {
  strategy.componentSelection {
    all {
      if (filter.transform(current)) {
        current.reject("Rejected")
      }
    }
  }
}
```

Reply:
oh, maybe not without losing the typesafe checking